apps/omg_watcher/lib/omg_watcher/api/status.ex

```diff
@@ -40,23 +41,33 @@ defmodule OMG.Watcher.API.Status do
   @decorate measure_event()
   @spec get_status() :: {:ok, t()} | {:error, atom}
   def get_status do
-    with {:ok, last_mined_child_block_number} <- Eth.RootChain.get_mined_child_block(),
-         {:ok, {_root, last_mined_child_block_timestamp}} <-
-           Eth.RootChain.get_child_chain(last_mined_child_block_number),
-         {:ok, child_block_interval} <- Eth.RootChain.get_child_block_interval() do
-      {state_current_block, _} = State.get_status()
+    with {:ok, child_block_interval} <- Eth.RootChain.get_child_block_interval(),
+         {:ok, mined_child_block_number} <- Eth.RootChain.get_mined_child_block(),
+         {:ok, {_root, mined_child_block_timestamp}} <- Eth.RootChain.get_child_chain(mined_child_block_number),
+         {state_current_block, _} = State.get_status(),
+         validated_child_block_number = state_current_block - child_block_interval,
+         {:ok, {_root, validated_child_block_timestamp}} <- Eth.RootChain.get_child_chain(validated_child_block_number),
+         {:ok, services_synced_heights} <- RootChainCoordinator.get_ethereum_heights(),
+         {:ok, eth_block_number} <- Eth.get_ethereum_height(),
+         {:ok, eth_block_timestamp} <- Eth.get_block_timestamp_by_number(eth_block_number) do
+      contract_addr = Eth.Diagnostics.get_child_chain_config()[:contract_addr] |> Eth.Encoding.from_hex()
 
       {_, events_processor} = ExitProcessor.check_validity()
       {_, events_block_getter} = BlockGetter.get_events()
       {:ok, in_flight_exits} = ExitProcessor.get_active_in_flight_exits()
 
       status = %{
-        last_validated_child_block_number: state_current_block - child_block_interval,
-        last_mined_child_block_number: last_mined_child_block_number,
-        last_mined_child_block_timestamp: last_mined_child_block_timestamp,
+        last_validated_child_block_number: validated_child_block_number,
+        last_validated_child_block_timestamp: validated_child_block_timestamp,
+        last_mined_child_block_number: mined_child_block_number,
+        last_mined_child_block_timestamp: mined_child_block_timestamp,
+        last_seen_eth_block_number: eth_block_number,
+        last_seen_eth_block_timestamp: eth_block_timestamp,
         eth_syncing: Eth.syncing?(),
         byzantine_events: events_processor ++ events_block_getter,
-        in_flight_exits: in_flight_exits
+        in_flight_exits: in_flight_exits,
+        contract_addr: contract_addr,
+        services_synced_heights: services_synced_heights
       }
 
       {:ok, status}
```

Comment:
if a `with` statement fails, you'll not be able to debug this.

Reply:
true that, probably the original `with` was misused here.

Shall we pattern match the `:ok`s and fail the request if the services relied upon fail?

Reply:
On second thought, I think it would be appropriate and consistent to return `{:error, ...}`, when the external services relied upon don't behave well.

But I agree, that the number of these calls would make the usefulness of the `with`-driven return low. How about we chunk it into some logical groups and decorate the final `{:error, ...}` returned with some useful information about which part specifically failed. Lemme propose some solution here.

Reply:
A simple but effective approach is that you wrap the right hand side in a tuple, for example:
`with {:get_user, {:ok, user}} <- {:get_user, User.get_user(id)}`
because when your right hand side fails, you know which clause failed and you can at least log that.

Reply:
eh, I can't seem to make up my mind today. 

For now I figured that all the things that can fail are services (nothing the user could fix), so I went with crashing the call. I pushed this version out now, but I can quite easily do the "structure `with`" back again.

Sorry to have complicated things.

Take a look how it is done now (care to take on the main review for this one, since we're already at it?)